portal-impl/src/com/liferay/portal/security/permission/ResourceActionsImpl.java

```diff
@@ -249,35 +250,37 @@ public String getModelResource(Locale locale, String name) {
 
 	@Override
 	public List<String> getModelResourceActions(String name) {
-		ResourceActionsBag modelResourceActionsBag = _getResourceActionsBag(
-			name, _modelResourceActionsBags);
+		ResourceActionsBagImpl modelResourceActionsBagImpl =
```

Comment:
@dantewang Please use interface rather that the concrete class in logic like this.